### PR TITLE
PC meta diffs

### PR DESCRIPTION
**Pull request by olsen232**
Detect changes to PC meta-items due to changes in PC meta-tiles,
add them to diff output. They can't currently be committed.

Codify something called a ListOfConflicts. Sort-of exists already
for metadata.xml in GPKGs, but it is now  much easier to create
a ListOfConflicts with Point Clouds, so it becomes a named Kart
concept that is easier to detect and outputs properly in diffs.

Fixed schema.py to support PC schemas that don't have an "id".

Still TODO:
- make PC diffs committable (when they don't have conflicts)
- improve the error messaging around ListOfConflicts
- more logic around formats (eg, allow committing a COPC dataset to
  a --preserve-format dataset)
- maybe improve the output of a ListOfConflicts in JSON

https://github.com/koordinates/kart/issues/565

### CODE REVIEW

File: kart/list_of_conflicts.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    - is not associated with any commit or commits
+    - is resolved by the user manually fixing the import source or working copy so that it has only one possibility.
+
+    By contrast, the conflicts from a merge - see AncestorOursTheirs in merge_util.py - are such:
+    - have two conflicting possibilities (or three if you count the ancestor as a third possibility)
+    - each conflicting possibility comes from a particular commit, where they were valid separately before the merge
+    - can be resolved by the user selecting which version will be the winner, by name eg ancestor, ours or theirs.
+    """
+
+    pass
```
Comment (craigds) [inline kart/list_of_conflicts.py:45]: minor: you don't need `pass` unless it's syntactically invalid without it. Here you've got a docstring so I reckon omit the `pass`, it doesn't add anything

File: kart/exceptions.py [hunk 2]
```diff
@@ -21,6 +21,8 @@
 SPATIAL_FILTER_PK_CONFLICT = 27
 INVALID_FILE_FORMAT = 28
 UNCOMMITTED_CHANGES = 29
+# Ran out of 2x numbers. Oh well.
+CONFLICTING_POSSIBILITIES = 31
```
Comment (craigds) [inline kart/exceptions.py:25]: I'm not really fond of this name, doesn't seem very descriptive of what it actually is. It's a conflict at commit time, maybe it shoudl be `AMBIGUOUS_COMMIT` or `WORKING_COPY_CONFLICT` or something

File: kart/list_of_conflicts.py [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+        for key, item in source.meta_items().items():
+            if isinstance(item, ListOfConflicts):
+                click.echo(
+                    f"Sorry, importing more than one {key} for a single dataset ({source.dest_path}) is not supported",
+                    err=True,
+                )
+                has_conflicts = True
+    if has_conflicts:
+        raise InvalidOperation(
+            "Failed to import into the Kart", exit_code=CONFLICTING_POSSIBILITIES
```
Comment (craigds) [inline kart/list_of_conflicts.py:80]:
```suggestion
            "Failed to import", exit_code=CONFLICTING_POSSIBILITIES
```

File: kart/list_of_conflicts.py (hunk 1 above)
Comment (olsen232) [reply kart/list_of_conflicts.py:45]: Done

File: kart/exceptions.py (hunk 2 above)
Comment (olsen232) [reply kart/exceptions.py:25]: Landed on WORKING_COPY_OR_IMPORT_CONFLICT. It's long but it works

File: kart/list_of_conflicts.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+        )
+
+
+def check_sources_are_importable(sources):
+    has_conflicts = False
+    for source in sources:
+        for key, item in source.meta_items().items():
+            if isinstance(item, ListOfConflicts):
+                click.echo(
+                    f"Sorry, importing more than one {key} for a single dataset ({source.dest_path}) is not supported",
```
[the remark this answers is not in the document]
Comment (olsen232) [reply kart/list_of_conflicts.py:74]: Done

File: kart/list_of_conflicts.py (hunk 3 above)
Comment (olsen232) [reply kart/list_of_conflicts.py:80]: Done

**Pull request merged**